packages/simcore-sdk/src/simcore_sdk/node_ports_common/file_io_utils.py

```diff
@@ -325,53 +328,62 @@ async def upload_file_to_presigned_links(
     file_chunk_size = int(file_upload_links.chunk_size)
     num_urls = len(file_upload_links.urls)
     last_chunk_size = file_size - file_chunk_size * (num_urls - 1)
-    upload_tasks = []
+
+    results: list[UploadedPart] = []
     async with AsyncExitStack() as stack:
         tqdm_progress = stack.enter_context(
             tqdm_logging_redirect(
                 desc=f"uploading {file_name}\n",
                 total=file_size,
                 **(
-                    _TQDM_FILE_OPTIONS
-                    | dict(miniters=_compute_tqdm_miniters(file_size))
+                    _TQDM_FILE_OPTIONS | {"miniters": _compute_tqdm_miniters(file_size)}
                 ),
             )
         )
         sub_progress = await stack.enter_async_context(
             progress_bar.sub_progress(steps=file_size)
         )
 
-        for index, upload_url in enumerate(file_upload_links.urls):
-            this_file_chunk_size = (
-                file_chunk_size if (index + 1) < num_urls else last_chunk_size
-            )
-            upload_tasks.append(
-                _upload_file_part(
-                    session,
-                    file_to_upload,
-                    index,
-                    index * file_chunk_size,
-                    this_file_chunk_size,
-                    upload_url,
-                    tqdm_progress,
-                    num_retries,
-                    io_log_redirect_cb=io_log_redirect_cb,
-                    progress_bar=sub_progress,
+        indexed_urls: list[tuple[int, AnyUrl]] = list(enumerate(file_upload_links.urls))
+        for slice_of_indexed_urls in slice_list_iter(
+            indexed_urls, slice_size=_CONCURRENT_MULTIPART_UPLOADS_COUNT
+        ):
+            upload_tasks = []
+            for index, upload_url in slice_of_indexed_urls:
+                this_file_chunk_size = (
```

Comment:
`this_file_chunk_size` is only used as input parameters in `_upload_file_part`. Why not just use the key arguments below i.e.
```python
   upload_tasks.append( 
   _upload_file_part(
    session=session,
    file_to_upload = file_to_upload,
    part_index= index,
    file_offset = index * file_chunk_size,
    file_part_size = file_chunk_size if  index<last_index else last_chunk_size ,
    upload_url=upload_url,
    pbar = tqdm_progress,
    num_retries = num_tries,
    io_log_redirect_cb - io_log_redirect_cb,
    progress_bar=sub_prgress,
    )
 )   
```